kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -11,7 +11,7 @@
 
 class ScatteringBase1D(ScatteringBase):
     def __init__(self, J, shape, Q=1, T=None, max_order=2, average=True,
-            oversampling=0, vectorize=True, out_type='array', backend=None):
```

Comment:
Why is the `out_type` here `array`, but default above is `dict`?

Reply:
for backwards compatibility. The current default is `out_type="array"` but `vectorize=False`, meaning, we don't return an array but a dict !!

We do
```python
if out_type == 'array' and not vectorize:
        out_S = {x['n']: x['coef'] for x in out_S}
```

which is extremely confusing and why i propose to remove the `vectorize` kwarg.

You might argue in favor of changing the default `out_type` to `"array"`, and i'd agree with you, but that would technically be a breaking change (even though all PT/TF/NP frontends override the `vectorize=False` default of `base_frontend` to make it `vectorize="true"`, and thus return an array)

I did not take the liberty to make this breaking change because i wanted to keep the PR minimal. Hope this clarifies

Reply:
Also note that `out_type="array"` as a default has some drawbacks, not the least of which being that it assumes `average` to be `True`.
passing `average=False` to this function and letting `out_type` be an array would cause an error at runtime, which we catch in the non-base frontends.

Perhaps the safest solution is no default value at all?